src/config/plugin.ts

```diff
@@ -112,6 +116,7 @@ export class Plugin implements IPlugin {
     this._debug('reading %s plugin %s', this.type, root)
     this.pjson = await loadJSON(path.join(root, 'package.json')) as any
     this.name = this.pjson.name
+    this.aliases = util.resolvePluginAliasNames(this.pjson)
```

Comment:
I think we can simplify this a lot by using `this.options.name` as the alias since that'll always be the name that given to the plugin by the cli:

```typescript
Plugin {
  options: {
    type: 'core',
    root: '/Users/mdonnalley/code/cli',
    name: '@sf/telemetry' // <= right here
  },
  _base: '@oclif/core@0.5.22',
  valid: false,
  alreadyLoaded: false,
  children: [],
  _debug: [Function (anonymous)],
  warned: false,
  type: 'core',
  tag: undefined,
  root: '/Users/mdonnalley/code/cli/node_modules/@sf/telemetry',
  pjson: {
    name: '@salesforce/plugin-telemetry',
    description: 'Command usage and error telemetry for the Salesforce CLI',
    version: '1.2.3',
    author: 'Salesforce',
    bugs: 'https://github.com/forcedotcom/cli/issues',
    dependencies: {
      '@oclif/config': '^1',
      '@salesforce/command': '^3.0.3',
      '@salesforce/core': '^2.24.0',
      '@salesforce/telemetry': '^2.0.4',
      tslib: '^2'
    },
    devDependencies: {
      '@oclif/dev-cli': '^1',
      '@oclif/plugin-command-snapshot': '^2.0.0',
      '@salesforce/cli-plugins-testkit': '^1.1.5',
      '@salesforce/dev-config': '^2.1.2',
      '@salesforce/dev-scripts': '0.9.15',
      '@salesforce/plugin-command-reference': '^1.3.0',
      '@salesforce/prettier-config': '^0.0.2',
      '@salesforce/ts-sinon': '1.3.12',
      '@types/debug': '^4.1.5',
      '@typescript-eslint/eslint-plugin': '^4.2.0',
      '@typescript-eslint/parser': '^4.2.0',
      chai: '^4.2.0',
      'cz-conventional-changelog': '^3.2.0',
      eslint: '^7.27.0',
      'eslint-config-prettier': '^8.3.0',
      'eslint-config-salesforce': '^0.1.6',
      'eslint-config-salesforce-license': '^0.1.6',
      'eslint-config-salesforce-typescript': '^0.2.7',
      'eslint-plugin-header': '^3.1.1',
      'eslint-plugin-import': '^2.20.2',
      'eslint-plugin-jsdoc': '^35.1.2',
      'eslint-plugin-prettier': '^3.1.3',
      husky: '^4.3.8',
      'lint-staged': '^11.0.0',
      mocha: '^8.4.0',
      nyc: '^15.1.0',
      prettier: '^2.0.5',
      'pretty-quick': '^3.1.0',
      shx: '0.3.3',
      sinon: '10.0.0',
      'ts-node': '^10.0.0',
      typescript: '^4.1.3'
    },
    config: { commitizen: [Object] },
    engines: { node: '>=12.0.0' },
    files: [ '/lib', '/processes', '/oclif.manifest.json' ],
    homepage: 'https://github.com/salesforcecli/plugin-telemetry',
    keywords: [ 'force', 'salesforce', 'sfdx', 'salesforcedx', 'sfdx-plugin' ],
    license: 'BSD-3-Clause',
    oclif: {
      commands: './lib/commands',
      bin: 'sfdx',
      hooks: [Object],
      devPlugins: [Array]
    },
    repository: 'salesforcecli/plugin-telemetry',
    scripts: {
      build: 'sf-build',
      clean: 'sf-clean',
      'clean-all': 'sf-clean all',
      'clean:lib': 'shx rm -rf lib && shx rm -rf coverage && shx rm -rf .nyc_output && shx rm -f oclif.manifest.json',
      compile: 'sf-compile',
      docs: 'sf-docs',
      format: 'sf-format',
      lint: 'sf-lint',
      postpack: 'shx rm -f oclif.manifest.json',
      posttest: 'yarn lint && yarn test:deprecation-policy && yarn test:command-reference',
      prepack: 'sf-prepack',
      prepare: 'sf-install',
      pretest: 'sf-compile-test',
      test: 'sf-test',
      'test:command-reference': './bin/run commandreference:generate --erroronwarnings',
      'test:deprecation-policy': './bin/run snapshot:compare',
      'test:nuts': 'nyc mocha "**/*.nut.ts" --slow 4500 --timeout 600000 --parallel',
      version: 'oclif-dev readme'
    },
    husky: { hooks: [Object] },
    publishConfig: { access: 'public' },
    sfdx: {
      publicKeyUrl: 'https://developer.salesforce.com/media/salesforce-cli/sfdx-cli-03032020.crt',
      signatureUrl: 'https://developer.salesforce.com/media/salesforce-cli/signatures/salesforce-plugin-telemetry-1.2.3.sig'
    }
  },
  name: '@salesforce/plugin-telemetry'
}
```

If we're able to rely on that, we're able to remove the `Alias` interface and the `resolvePluginAliasNames` function

Reply:
I totally agree with Mike here.

And shouldn't this just be `alias`? Or do we want to support installing the same plugin under different aliases? That just seems like asking for trouble to me.
Example:
```js
"dependencies": {
    "@sf/env": "npm:@salesforce/plugin-env@0.0.9",
	"@sf/env2": "npm:@salesforce/plugin-env@0.0.9",
  }
  ```